### PR TITLE
test: Omnichannel Users & Departments

### PR DESCRIPTION
**Pull request by KevLehman**
https://rocketchat.atlassian.net/browse/OC-1382

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/livechat/10-departments.ts [hunk 1]
```diff
@@ -18,6 +19,67 @@ import { updatePermission, updateSetting } from '../../../data/permissions.helpe
 import { createUser, deleteUser } from '../../../data/users.helper';
 import { IS_EE } from '../../../e2e/config/constants';
 
+(IS_EE ? describe.skip : describe)('LIVECHAT - Departments[CE]', () => {
+	before((done) => getCredentials(done));
+
+	before(async () => {
+		await updateSetting('Livechat_enabled', true);
+		await updatePermission('view-livechat-manager', ['admin']);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/10-departments.ts:27]:
```suggestion
		await restorePermissionToRoles('view-livechat-manager');
```

File: apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts [hunk 2]
```diff
@@ -70,6 +70,165 @@ type TestUser = { user: IUser; credentials: { 'X-Auth-Token': string; 'X-User-Id
...
+				})
+				.expect(200);
+
+			expect(body.success).to.be.true;
+		});
+	});
+
+	describe('[GET] livechat/monitors', () => {
+		it('should fail if manage-livechat-monitors permission is missing', async () => {
+			await updatePermission('manage-livechat-monitors', []);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts:193]:
```suggestion
			await removePermissionFromAllRoles('manage-livechat-monitors');
```

File: apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts [hunk 3]
```diff
@@ -70,6 +70,165 @@ type TestUser = { user: IUser; credentials: { 'X-Auth-Token': string; 'X-User-Id
...
+			return request.get(api('livechat/monitors')).set(credentials).expect(403);
+		});
+		it('should return all monitors', async () => {
+			await updatePermission('manage-livechat-monitors', ['admin']);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts:197]:
```suggestion
			await restorePermissionToRoles('manage-livechat-monitors');
```

File: apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts [hunk 4]
```diff
@@ -70,6 +70,165 @@ type TestUser = { user: IUser; credentials: { 'X-Auth-Token': string; 'X-User-Id
...
+			expect(body.monitors[0]).to.have.property('username', user.username);
+
+			// cleanup
+			await deleteUser(user);
+		});
+	});
+
+	describe('livechat/monitors/:username', () => {
+		it('should fail if manage-livechat-monitors permission is missing', async () => {
+			await updatePermission('manage-livechat-monitors', []);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts:213]:
```suggestion
			await removePermissionFromAllRoles('manage-livechat-monitors');
```

File: apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts [hunk 5]
```diff
@@ -70,6 +70,165 @@ type TestUser = { user: IUser; credentials: { 'X-Auth-Token': string; 'X-User-Id
...
+			return request.get(api('livechat/monitors/123')).set(credentials).expect(403);
+		});
+		it('should return a monitor', async () => {
+			await updatePermission('manage-livechat-monitors', ['admin']);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/22-monitors.ts:217]:
```suggestion
			await restorePermissionToRoles('manage-livechat-monitors');
```

**Review by murtaza98 — DISMISSED** [review]

File: apps/meteor/tests/end-to-end/api/livechat/10-departments.ts [hunk 6]
```diff
@@ -76,8 +204,62 @@ import { IS_EE } from '../../../e2e/config/constants';
...
+						showOnRegistration: true,
+						email: 'bla@bla',
+						requestTagBeforeClosingChat: true,
+					},
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400);
+		});
+
+		it('should return an error if requestTagsBeforeClosing is true but no tags are not an array', async () => {
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/10-departments.ts:225]:
```suggestion
		it('should return an error if requestTagsBeforeClosing is true but tags are not an array', async () => {
```

Comment (KevLehman) [reply apps/meteor/tests/end-to-end/api/livechat/10-departments.ts:225]: image

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**